packages/api/src/resolvers/subscriptions/index.ts

```diff
@@ -203,49 +186,83 @@ export const subscribeResolver = authorized<
       },
     })
 
-    // create new rss subscription
-    if (input.url) {
-      const MAX_RSS_SUBSCRIPTIONS = 150
-      // validate rss feed
-      const feed = await parser.parseURL(input.url)
-
-      // limit number of rss subscriptions to 50
-      const newSubscriptions = (await authTrx((t) =>
-        t.query(
-          `insert into omnivore.subscriptions (name, url, description, type, user_id, icon) 
-        select $1, $2, $3, $4, $5, $6 from omnivore.subscriptions 
-        where user_id = $5 and type = 'RSS' and status = 'ACTIVE' 
-        having count(*) < $7 
-        returning *;`,
-          [
-            feed.title,
-            input.url,
-            feed.description || null,
-            SubscriptionType.Rss,
-            uid,
-            feed.image?.url || null,
-            MAX_RSS_SUBSCRIPTIONS,
-          ]
-        )
-      )) as Subscription[]
-
-      if (newSubscriptions.length === 0) {
+    // find existing subscription
+    const existingSubscription = await getRepository(Subscription).findOneBy({
+      url: input.url,
+      user: { id: uid },
+      type: SubscriptionType.Rss,
+    })
+    if (existingSubscription) {
+      if (existingSubscription.status === SubscriptionStatus.Active) {
         return {
-          errorCodes: [SubscribeErrorCode.ExceededMaxSubscriptions],
+          errorCodes: [SubscribeErrorCode.AlreadySubscribed],
```

Comment:
Will probably try to handle this in the client in the future to give a useful message.

Reply:
The Web client handles the error code and gives some meaningful error messages for subscriptions